tesseract/tesseract_planning/tesseract_process_managers/src/process_planning_server.cpp

```diff
@@ -0,0 +1,486 @@
+{
+  DescartesTaskflowParams global_params;
+  global_params.enable_simple_planner = isNullInstruction(request.seed);
+  global_params.enable_post_contact_discrete_check = false;
+  global_params.enable_post_contact_continuous_check = false;
+  global_params.enable_time_parameterization = false;
+  global_params.profiles = profiles;
+  GraphTaskflow::UPtr global_task = createDescartesTaskflow(global_params);
+
+  TrajOptTaskflowParams raster_params;
+  raster_params.enable_simple_planner = false;
+  raster_params.profiles = profiles;
+  GraphTaskflow::UPtr raster_task = createTrajOptTaskflow(raster_params);
+  GraphTaskflow::UPtr transition_task = createTrajOptTaskflow(raster_params);
+
+  return std::make_unique<RasterOnlyGlobalTaskflow>(
+      std::move(global_task), std::move(transition_task), std::move(raster_task));
+}
+
+ProcessPlanningServer::ProcessPlanningServer(EnvironmentCache::Ptr cache, size_t n)
+  : cache_(std::move(cache)), executor_(std::make_shared<tf::Executor>(n))
+{
+  registerProcessPlanner(process_planner_names::TRAJOPT_PLANNER_NAME, &createTrajOptGenerator);
+  registerProcessPlanner(process_planner_names::OMPL_PLANNER_NAME, &createRasterGenerator);
+  registerProcessPlanner(process_planner_names::DESCARTES_PLANNER_NAME, &createRasterGenerator);
+  registerProcessPlanner(process_planner_names::CARTESIAN_PLANNER_NAME, &createRasterGenerator);
+  registerProcessPlanner(process_planner_names::FREESPACE_PLANNER_NAME, &createRasterGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_FT_PLANNER_NAME, &createRasterGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_O_FT_PLANNER_NAME, &createRasterOnlyGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_G_FT_PLANNER_NAME, &createRasterGlobalGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_FT_DT_PLANNER_NAME, &createRasterDTGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_FT_WAAD_PLANNER_NAME, &createRasterWAADGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_FT_WAAD_DT_PLANNER_NAME, &createRasterWAADDTGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_O_G_FT_PLANNER_NAME, &createRasterOnlyGlobalGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_CT_PLANNER_NAME, &createRasterCTGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_O_CT_PLANNER_NAME, &createRasterOnlyCTGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_CT_DT_PLANNER_NAME, &createRasterCTDTGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_CT_WAAD_PLANNER_NAME, &createRasterCTWAADGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_CT_WAAD_DT_PLANNER_NAME, &createRasterCTWAADDTGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_G_CT_PLANNER_NAME, &createRasterGlobalCTGenerator);
+  registerProcessPlanner(process_planner_names::RASTER_O_G_CT_PLANNER_NAME, &createRasterOnlyGlobalCTGenerator);
```

Comment:
Do you think we should do this? How big are these objects? In most cases a user is only going to be using a handful of these.

Reply:
How about moving them to a method called loadDefaultProcessPlanners()? The user can decide to call it or not.

Reply:
Yeah that works.

Reply:
Just pushed the change.